synapse/lib/ast.py

```diff
@@ -4148,6 +4152,89 @@ async def feedfunc():
             async for item in agen:
                 yield item
 
+class CondSetOper(Oper):
+    def __init__(self, astinfo, kids, errok=False):
+        Value.__init__(self, astinfo, kids=kids)
+        self.errok = errok
+
+    def prepare(self):
+        self.isconst = False
+        if isinstance(self.kids[0], Const):
+            self.isconst = True
+            match self.kids[0].value():
+                case "unset":
+                    self.valu = SET_UNSET
+
+    async def compute(self, runt, path):
+        if self.isconst:
+            return self.valu
+
+        valu = await self.kids[0].compute(runt, path)
+        match valu:
+            case "always":
+                return SET_ALWAYS
+            case "unset":
+                return SET_UNSET
+            case "never":
+                return SET_NEVER
+            case _:
+                mesg = f'Invalid conditional set operator ({valu}).'
+                exc = s_exc.StormRuntimeError(mesg=mesg)
+                raise self.addExcInfo(exc)
+
+class EditCondPropSet(Edit):
+
+    async def run(self, runt, genr):
+
+        if runt.readonly:
+            mesg = 'Storm runtime is in readonly mode, cannot create or edit nodes and other graph data.'
+            raise self.addExcInfo(s_exc.IsReadOnly(mesg=mesg))
+
+        excignore = (s_exc.BadTypeValu,) if self.kids[1].errok else ()
+        rval = self.kids[2]
+
+        async for node, path in genr:
+
+            propname = await self.kids[0].compute(runt, path)
+            name = await tostr(propname)
+
+            prop = node.form.props.get(name)
```

Comment:
This could use `Form.reqProp()` for consistency

Reply:
Yeah this does a little extra checking and adds error highlighting info but we could add that into `reqProp()`

Reply:
Actually I'm not sure we want to move the additional type checking we do here into the datamodel

Reply:
Updated to use reqProp and add an `extra` arg there.